sdccc/src/main/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ConditionalPreconditions.java

```diff
@@ -688,6 +726,38 @@ static boolean manipulation(final Injector injector) {
         }
     }
 
+    /**
+     * Precondition which triggers all possible description changes with parent-child relationships.
+     */
+    public static class DescriptionModificationAllWithParentChildRelationshipPrecondition extends SimplePrecondition {
+
+        private static final Logger LOG = LogManager.getLogger(DescriptionModificationAllWithParentChildRelationshipPrecondition.class);
+
+        /**
+         * Creates a description changed precondition check.
+         */
+        public DescriptionModificationAllWithParentChildRelationshipPrecondition() {
+            super(DescriptionModificationAllWithParentChildRelationshipPrecondition::preconditionCheck, DescriptionModificationAllWithParentChildRelationshipPrecondition::manipulation);
+        }
+
+        static boolean preconditionCheck(final Injector injector) {
+            return false; // always trigger description changes
```

Comment:
If there is no condition as to when the manipulation should be executed, then I would suggest moving the manipulation to ManipulationsPrecondition.

Reply:
This would save some LOCs (the preconditionCheck, mainly) and makes sense from a consistency point of view, but would make it neccessary to duplicate the 2 methods descriptionModificationManipulation() and descriptionUpdateWithParentChildRelationshipManipulation() both in ManipulationPreconditions...
which one do you consider worse?

Reply:
I would prefer consistency, plus the RemoveAndReinsertDescriptor modification in ManipulationPreconditions has almost the exact same structure as `descriptionModificationManipulation`.